earth_enterprise/BUILD_RHEL_Centos.md

```diff
@@ -0,0 +1,133 @@
+# GEE Build Setup for CentOS, RHEL 6 and RHEL 7     
+
+
+## Install git  
+    
+Install the system default version:
+    
+```
+sudo yum install git
+```
+    
+Or install the latest version from the IUS repo (recommended) [ [More Info] ](https://ius.io/GettingStarted/):
+```
+sudo yum install -y wget
+cd /tmp
+wget https://rhel7.iuscommunity.org/ius-release.rpm
+sudo yum install ius-release.rpm
+sudo yum install git2u-all
+```
+        
+        
+## Install git lfs
+   
+Execute the following commands
+    
+```
+curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.rpm.sh | sudo bash
+sudo yum install git-lfs
+```
+
+
+
+## Enable additional repos
+Different yum repositoies need to be enabled for different distributions to install several required development tools.  
+
+### CentOS 7 
+```
+sudo yum install epel-release
+```
+
+### RHEL 7  
+
+_note:_ If the first command doesn't work, try the alternate methods listed [here](https://www.cyberciti.biz/faq/installing-rhel-epel-repo-on-centos-redhat-7-x/)
+```
+sudo subscription-manager repos --enable=rhel-7-server-optional-rpms
+sudo subscription-manager repos --enable=rhel-7-server-optional-source-rpms
+```    
+
+### RHEL 6
+```
+sudo subscription-manager repos --enable=rhel-x86_64-workstation-dts-2
+sudo subscription-manager repos --enable=rhel-6-server-optional-rpms
+sudo subscription-manager repos --enable=rhel-6-server-optional-source-rpms
+```
+       
+## GCC 4.8
+
+### CentOS 7 and RHEL 7
+
+To obtain GCC 4.8 on RHEL6, you must enable the "Development Tools" repository in the RHSM: 
+```
+sudo yum --setopt=group_package_types=mandatory,default,optional groupinstall "Development Tools"
+sudo yum install scons perl-Perl4-CoreLibs xorg-x11-server-devel python-devel perl-Alien-Packages gtest-devel openssl-devel libxml2-devel libXinerama-devel libXft-devel libXrandr-devel libXcursor-devel gdbm-devel libmng-devel libcap-devel libpng12-devel libXmu-devel freeglut-devel zlib-devel libX11-devel bison-devel openjpeg-devel openjpeg2-devel geos-devel proj-devel ogdi-devel giflib-devel xerces-c xerces-c-devel
+```
+### RHEL 6
+
+```
+sudo yum install devtoolset-2-gcc devtoolset-2-binutils  devtoolset-2-gcc-c++ devtoolset-2-gcc-gfortran
+```
+
+The GCC 4.8 installation will be located in the `/opt/rh/devtoolset-2/root/usr/bin/` directory.  
+
+The GEE build scripts will detect this compiler automatically. However, if you wish switch the environment to use GCC 4.8 in a shell, run: 
+```source /opt/rh/devtoolset-2/enable```
+
+## GTest 1.8
+
+
+### CentOS 7 and RHEL 7
+
+gtest is included in EPEL and RHEL Extra Repositories.  Install the RPM: 
+```
+yum install gtest-devel
+``` 
+
+### RHEL 6
+You will need to compile, package and install an updated version of GTest as an RPM for RHEL6.   Note that this build process also depends on GCC 4.8, as does the rest of the GEE build process. 
```

Comment:
Would there be any benefit to providing a pre-compiled rpm?

Reply:
For some customers, yes.  It minimizes the complexity of the build by separation of concerns, and allows any GTest updates to be handled independently.  It also keeps the build more aligned between newer and older operating systems.    More thoughts?

Reply:
I think, if we know what processors to compile for we should provide packages. We could add them as files that are part of a release of the RPM packaging repository so they're online. It would be even easier, if we have a suitable Yum repository, but that may involve more details.

I haven't looked at the license too closely, but as far as I remember Google Test is under BSD, so there shouldn't be a problem with deriving a package from it, and distributing it.

Reply:
@tst-ppenev  - this would be another story though, right? (multi-arch builds?)

Reply:
If we want multi-architecture builds, yeah. We should be able to cross-compile, or maybe we have the architectures to run on.

We can write some stories, depending what is necessary.